<!-- -->
<launch>

	<arg name="base_control_port" default="/dev/U5"/>   <!--底层can分析仪-->
	<arg name="gps_port" default="/dev/ttyS2"/>		    <!-- GPS-->
	<arg name="stm32_port" default="/dev/U6"/>          <!-- STM32-->
	
	<arg name="pathTracking_file_name" default="$(find data)/data/_cycle.txt"/>
	
<!--底层 -->	
	<node name="base_control" pkg="little_ant" type="base_control" output="screen">
		<param name ="obd_can_port_name" value="$(arg base_control_port)"/>
		<param name ="stm32_port_name" value="$(arg stm32_port)"/>
		<param name ="stm32_baudrate" value="115200"/>
		<param name ="max_steering_speed" value="2.0"/> <!--前后帧转角最大差值 -->
		
	</node>

<!--避障-->	
	<node type="avoiding_node" pkg="avoiding"  name="avoiding_node" output="screen" >

		<param name="objects_topic" value="/detection/lidar_objects" />
		
		<param name="path_points_file" value="$(arg pathTracking_file_name)"/>
		
		<param name="safety_distance_side" value="0.4"/>    <!--   两侧安全距离       -->
		<param name="danger_distance_side" value="0.25"/>
		
		<param name="max_deceleration" value="5.0" />
		<param name="maxOffset_left" value="-3.0"/>  <!-- 左右侧避障偏移最大值 -->
		<param name="maxOffset_right" value="0.0"/>
		
	</node>

<!-- 跟踪   -->
	<node name="path_tracking_node" pkg="path_tracking" type="path_tracking_node" output="screen">
		<param name="path_points_file" value="$(arg pathTracking_file_name)"/>
		<param name="disThreshold" value="6.0"/>    			<!-- 跟踪前视距离  -->
		<param name="avoiding_disThreshold" value="15.0"/>		<!-- 跟踪前视距离  -->
		<param name="speed" value="20.0"/>						<!-- 跟踪速度  -->
		<param name="maxOffset_left" value="-3.0"/>  <!-- 左右侧避障偏移最大值 -->
		<param name="maxOffset_right" value="0.0"/>
		<param name="foreSightDis_speedCoefficient" value = "1.8"/>
		<param name="foreSightDis_latErrCoefficient" value = "0.3"/>
		
	</node>
	
	
	
<!--GPS-->	
	<node pkg="novatel" type="novatel_node" name="novatel_node"  >

		<param name="port" value="$(arg gps_port)" />
		<param name="baudrate" value="115200" />
		<param name="odom_topic" value="/gps_odom" />
		<!-- <param name="log_commands" value="inspvaxb ontime 0.05" />-->
		<param name="log_commands" value="inspvaxb ontime 0.05; bestutmb ontime 0.05" />
		<!-- <param name="configure_port" value="COM2,9600,RTCM,NONE" /> -->
		<!-- <param name="log_commands" value="BESTUTMB ONTIME 1.0; BESTVELB ONTIME 1.0" /> -->
		<param name="gps_default_logs_period" value="0.0" />
		<param name="span_default_logs_period" value="0.0" />
		<param name="inspvax_topic" value = "/gps" />
	</node>
	
<!-- lat,lon,yaw + x,y = utm_gps -->	
	<node pkg="utm_gps" type="utm_gps" name="utm_gps"/>
	

<!--调试-->	
	<node name="record_gps_data_node" pkg="record_gps_data" type="record_gps_data_node">
		<param name="file_path" value="$(find data)/data/debug.txt"/>
	</node>

 <!--总决策-->
	<node  type="decision_making_node" pkg ="decision_making" name="decision_making_node" >
		<param name ="traffic_mark_topic" value="/traffic_mark" />
	</node>  
	
	

</launch>



